xarray/core/dataset.py

```diff
@@ -7074,5 +7074,51 @@ def query(
         # apply the selection
         return self.isel(indexers, missing_dims=missing_dims)
 
+    def drop_duplicates(
+        self,
+        subset: Optional[Union[Hashable, Sequence[Hashable]]] = None,
+        keep: Union[str, bool] = "first"
+    ):
+        """Returns a new dataset with duplicate dimension values removed.
+
+        Parameters
+        ----------
+        subset : dimension label or sequence of labels, optional
+            Only consider certain dimensions for identifying duplicates, by
+            default use all of the columns.
+        keep : {"first", "last", False}, default: "first"
+            Determines which duplicates (if any) to keep.
+            - ``first`` : Drop duplicates except for the first occurrence.
+            - ``last`` : Drop duplicates except for the last occurrence.
+            - False : Drop all duplicates.
+
+        Returns
+        -------
+        Dataset
+        """
+        if subset is None:
+            subset = list(self.coords)
+        elif isinstance(subset, str):
+            subset = [subset]
+
+        for dim in subset:
+            if dim not in self.dims:
+                raise ValueError("%s must be a single dataset dimension" % dim)
+
+        new = self.copy(deep=False)
+        if len(subset) > 1:
+            new = self.stack({'tmp_dim': subset})
```

Comment:
This could have fairly poor performance. Could we iterate through the dims / coords and drop for each?